packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -255,9 +256,49 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
 		super.startSummary(referenceSequenceNumber, summaryLogger);
 	}
 
+	/**
+	 * Validates that the in-progress summary is correct for all nodes, i.e., GC should have run for non-skipped nodes.
+	 * @param parentSkipRecursion - true if the parent of this node skipped recursing the child nodes when running GC.
+	 * In that case, the children will not have work-in-progress state.
+	 *
+	 * @returns ValidateSummaryResult which contains a boolean success indicating whether the validation was successful.
+	 * In case of failure, additional information is returned indicating type of failure and where it was.
+	 */
+	protected validateSummaryCore(parentSkipRecursion: boolean): ValidateSummaryResult {
+		/**
+		 * The absence of wip used routes indicates that GC was not run on this node. This can happen if:
+		 * 1. A child node was created after GC was already run on the parent. For example, a data store
+		 * is realized (loaded) after GC was run on it creating summarizer nodes for its DDSes. In this
+		 * case, the used routes of the parent should be passed on the child nodes and it should be fine.
```

Comment:
> the used routes of the parent should be passed on the child nodes and it should be fine

Meaning, `this.wipSerializedUsedRoutes` would _not_ be `undefined`?  If so then I'd restructure this comment block slightly to make that clearer.

Reply:
done.